backend/plan/views.py

```diff
@@ -176,6 +183,106 @@ def recommend_courses_view(request):
+        user = request.user
+        friend = request.data.friend_id
+
+        primary_schedule = self.queryset.get(person=user)
+        if primary_schedule:
+            res["primary_schedule"] = model_to_dict(primary_schedule)
+        else:   
+            res["primary_schedule"] = {}
+
+        if friend: 
+            if (not Friendship.are_friends(user.id, friend)):
+                res['message'] = "User is not friends with specified friend"
+                return JsonResponse(res, status=status.HTTP_400_BAD_REQUEST)
+            schedule = self.queryset.get(person=friend).schedule
+            if not schedule:
+                res["message"] = "No primary schedule found for this friend"
+                return JsonResponse(res, status=status.HTTP_404_NOT_FOUND)
+            # return the schedule
+            res = model_to_dict(schedule)
+            res["message"] = "Friend's primary schedule retrieved."
+            return JsonResponse(res)
+        else: 
+            # if no friend is specified, return all shared schedules with the current user
+            all_friends = get_accepted_friends(user)
+            friends = []
+            for friend in all_friends:
+                # get the friend's id from the list of friendship models for this user
+                query_friend = friend.sender if friend.sender != user else friend.recipient
+                schedule = self.queryset.get(person=query_friend).schedule
+                if schedule:
+                    friends.append( (query_friend, model_to_dict(schedule)) )  
+                else:
+                    # no primary schedule associated with this friend
+                    friends.append( (query_friend, {}) ) 
+                    
+            res["message"] = "All primary schedules of friends retrieved."
+            res["schedules"] = friends
+            return JsonResponse(res)
+
+    def post(self, request):
+        res = {}
+        # verify that schedule exist and set it as the primary for the passed in user
+
+        user = request.user
+        schedule = get_object_or_404(Schedule, pk=request.data.schedule_id)
+        if not schedule:
+            res["message"] = "Schedule does not exist"
+            return JsonResponse(res, status=status.HTTP_STATUS_400_BAD_REQUEST)
+
+        # check syntax for this but ensure that the schedule belongs to the user
+        if schedule.person != user:
+            res["message"] = "Schedule does not belong to user"
+            return JsonResponse(res, status=status.HTTP_STATUS_403_FORBIDDEN)
+
+        # set the schedule as the primary schedule for the user (and return old shared schedule if exists)
+        primary_schedule_entry = self.queryset.get(person=user)
+        if primary_schedule_entry:
+            res['old_shared_schedule'] = model_to_dict(primary_schedule_entry.schedule)
+            res['new_shared_schedule'] = model_to_dict(schedule)
```

Comment:
I honestly don't think we need to respond with this information. Just setting `res["message"]` here is probably fine.